### DISCUSSION TITLE
Battery Saving Deep Sleep with I2c VL53L0X

### DISCUSSION THREAD

**Discussion by zissouz** [arendst/Tasmota#13531] [Q&A] [answered]
Hi i am trying to monitor liquid levels in a large external container using VL53L0X. I have managed to compile the software with all the switches i need to work for the main purpose, but i have noticed that while the sensor shutdown pins are understood and mapped, they are only done so to support multiple sensors (to change addresses). 

I note from the documentation i can use  case FUNC_SAVE_BEFORE_RESTART: in the driver to disable the sensor, but there is quite a lot of code to handle addressing and pins that exceeds my understanding.

https://github.com/arendst/Tasmota/blob/development/tasmota/xsns_45_vl53l0x.ino

Is anyone able to point me in the right direction on this:

Am i missing an easy way to incorporate  FUNC_SAVE_BEFORE_RESTART into this driver given that it already seems to be doing the hard work of identifying the xshut  pins from the template?
or
Is it possible to use a rule to set a gpio pin low on System#Save - this will ensure the sensor is always shutdown? 
or
Is there a tasmota pin that will always go low on deepsleep
or...
Is there a better approach?

Any pointers appreciated but not expected. Thank you.

**Comment by sfromis** [comment, marked as answer]
Maybe explain what problem you're trying to deal with. Normally, the I2C sensors can work across deep sleep restarts, without needing to mess around in the drivers, or disabling sensors.

To use such a sensor with deep sleep, I'd want to not at all have power on the sensor while the MCU is effectively off, and instead of supplying permanent power to the sensor, use a gpio configured as "Output Hi", to supply power when Tasmota boots. Maybe two, depending on power drain. I've not tested if this works with this sensor, but expect chances to be pretty good.

**Comment by zissouz** [comment]
For clarity i am trying to extend battery life. I need it to wake up a few times a day, post a value via MQTT and the rest of the time it will be off. I have a board with possibly unreliable but bold claims of deep sleep power consumption (dfrobot) - its working fine for the sensor etc, but I am now moving onto the deep sleep function, and I dont have the kit to measure the consumption.

In terms of your suggestion...that sounds a lot simpler - thanks for getting back so quickly. I will give it a go. 

I think i got carried away with the product sheet that claims 5uA for hardware standby using xshut, but you are right...i dont need any sort of rapid response times or anything like that so there is no need to try and invoke standby modes.

Thanks again - just the sort of response i needed to help me move on in the right direction.
